relay-ourlogs/src/vercel_to_sentry.rs

```diff
@@ -0,0 +1,722 @@
+//! Transforms the Vercel Log Drain format to Sentry Logs.
+
+use std::str::FromStr;
+
+use chrono::{TimeZone, Utc};
+use relay_conventions::{ENVIRONMENT, HTTP_RESPONSE_STATUS_CODE, ORIGIN, SERVER_ADDRESS, URL_PATH};
+use relay_event_schema::protocol::{Attributes, OurLog, OurLogLevel, SpanId, Timestamp, TraceId};
+use relay_protocol::{Annotated, Meta, Remark, RemarkType};
+use serde::{Deserialize, Serialize};
+
+/// Log Level for Vercel Logs
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum VercelLogLevel {
+    /// Info level logs.
+    Info,
+    /// Warning level logs.
+    Warning,
+    /// Error level logs.
+    Error,
+    /// Fatal level logs.
+    Fatal,
+}
+
+/// Vercel proxy information for requests.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct VercelProxy {
+    /// Unix timestamp when the proxy request was made.
+    pub timestamp: i64,
+    /// HTTP method of the request.
+    pub method: String,
+    /// Hostname of the request.
+    pub host: String,
+    /// Request path with query parameters.
+    pub path: String,
+    /// User agent strings of the request.
+    #[serde(rename = "userAgent")]
+    pub user_agent: Vec<String>,
+    /// Referer of the request.
+    pub referer: String,
+    /// Region where the request is processed.
+    pub region: String,
+    /// HTTP status code of the proxy request.
+    #[serde(rename = "statusCode")]
+    pub status_code: Option<i64>,
+    /// Client IP address.
+    #[serde(rename = "clientIp")]
+    pub client_ip: Option<String>,
+    /// Protocol of the request.
+    pub scheme: Option<String>,
+    /// Size of the response in bytes.
+    #[serde(rename = "responseByteSize")]
+    pub response_byte_size: Option<i64>,
+    /// Original request ID when request is served from cache.
+    #[serde(rename = "cacheId")]
+    pub cache_id: Option<String>,
+    /// How the request was served based on its path and project configuration.
+    #[serde(rename = "pathType")]
+    pub path_type: Option<String>,
+    /// Variant of the path type.
+    #[serde(rename = "pathTypeVariant")]
+    pub path_type_variant: Option<String>,
+    /// Vercel-specific identifier.
+    #[serde(rename = "vercelId")]
+    pub vercel_id: Option<String>,
+    /// Cache status sent to the browser.
+    #[serde(rename = "vercelCache")]
+    pub vercel_cache: Option<String>,
+    /// Region where lambda function executed.
+    #[serde(rename = "lambdaRegion")]
+    pub lambda_region: Option<String>,
+    /// Action taken by firewall rules.
+    #[serde(rename = "wafAction")]
+    pub waf_action: Option<String>,
+    /// ID of the firewall rule that matched.
+    #[serde(rename = "wafRuleId")]
+    pub waf_rule_id: Option<String>,
+}
+
+/// Vercel log structure matching their schema.
+/// Based on: https://vercel.com/docs/drains/reference/logs
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct VercelLog {
```

Comment:
nit: I would prefer if this was the first type definition in the file (order from high-level to low-level).